tests/queue/tests.c

```diff
@@ -0,0 +1,411 @@
+/* Copyright 2014 Nodalink EURL
+
+ * This file is part of Butterfly.
+ *
+ * Butterfly is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published
+ * by the Free Software Foundation.
+ *
+ * Butterfly is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butterfly.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <glib.h>
+#include <string.h>
+#include <packetgraph/packetgraph.h>
+#include "brick-int.h"
+#include "packets.h"
+#include "utils/mempool.h"
+#include "utils/bitmask.h"
+#include "utils/mac.h"
+#include "collect.h"
+
+#define CHECK_ERROR(error) do {                 \
+               if (error) {                     \
+                       pg_error_print(error);   \
+		       pg_error_free(error);    \
```

Comment:
I'm not sure it's really useful to free error, knowing that we're going to abort at the next call.